paimon-core/src/main/java/org/apache/paimon/manifest/ManifestFileMeta.java

```diff
@@ -280,31 +299,65 @@ public static List<ManifestFileMeta> tryFullCompaction(
                     }
                 }
             } else {
-                // There is no DELETE Entry in Delta,Base don't need compaction
+                // There is no DELETE Entry in Delta, Base don't need compaction
                 j = base.size();
                 result.addAll(base);
             }
         }
 
-        Map<ManifestEntry.Identifier, ManifestEntry> fullMerged = new LinkedHashMap<>();
+        // 2.2. try to skip base files by reading entries
+
```

Comment:
Do we need to add a condition if (partitionType.getFieldCount() ==0)

Reply:
I feel also do not need, now this to check the cost should not be high, just hashcode to compare